rust/sealevel/client/src/router.rs

```diff
@@ -540,6 +541,16 @@ fn configure_owner(
     let expected_owner = Some(router_config.ownable.owner(ctx.payer_pubkey));
 
     if actual_owner != expected_owner {
+        if actual_owner != Some(ctx.payer_pubkey) {
+            println!(
+                "WARNING: Ownership transfer cannot be completed for chain: {} ({}) from {:?} to {:?}, the existing owner is not the payer account",
+                chain_config.name,
+                chain_config.domain_id(),
+                actual_owner,
+                expected_owner,
+            );
+            return;
```

Comment:
If we exit `configure_owner` here, we don't need to run `configure_upgrade_authority`, correct? I wonder if we can  return some data here so that we can decide if we need to run `configure_upgrade_authority`.